### PR TITLE
fix(gui-app): require a real token boundary across a node split

### PR DESCRIPTION
**Pull request by hdkshingala**
Follow-up to #898, which merged while this was being reviewed.

## The bug

#898 added a cross-node guard so a split run could not chip a prefix. It asked whether the next sibling **continues the command name**:

```ts
function continuesCommandName(text: string | null): boolean {
  return text !== null && /^[A-Za-z0-9:_-]/.test(text);
}
```

Punctuation continues no name, so it slipped straight through. A bold `$review` beside a plain `.` chipped:

| nodes | chipped? | serialized prompt |
|---|---|---|
| `"$review."` (one node) | no | `$review.` — stays prose |
| `["$review", "."]` (split) | **yes** | `/review.` |

Both should be prose. `LEADING_SLASH_COMMAND_REGEX` requires whitespace-or-end after the name, and `$review.` fails that — verified directly:

```
"$review."      no match
"$review"       MATCH name=review
"$review now"   MATCH name=review
```

The host's `parseProviderSlashPrompt` refuses `/review.` for the same reason. So the split case emitted a **structural** invocation of `review` that neither parser would have produced from the text — a command the user never wrote, running silently.

## The fix

Ask whether the sibling **opens a boundary**, not whether it continues the name. Those answers differ exactly on punctuation, which continues no name yet closes no token either.

```ts
function closesLeadingToken(node: JsonContent | undefined): boolean {
  if (node === undefined) return true;              // end of inline content
  if (node.type === "hardBreak") return true;       // serializes to a newline
  return node.type === "text" && /^\s/.test(node.text ?? "");
}
```

This also covers cases the old predicate never considered — a mention chip directly after the token (`$review@src/foo.ts`) no longer counts as a boundary.

## Tests

Two new, both mutation-probed:

- `does not chip a leading token an adjacent node ends with punctuation` — red under the old `continuesCommandName` predicate
- `chips a leading token when the adjacent node opens with whitespace` — red if `closesLeadingToken` is stubbed to refuse everything, so the fix cannot pass by simply never chipping across a node split

The `-code` case from #898 stays covered: `-` is a name character, so it opens no boundary either.

Full composer suites: 546 passing.

Found by Codex on #898 ([thread](https://github.com/traycerai/traycer/pull/898#discussion_r3699175000)).

### CODE REVIEW

**Comment by hdkshingala** [comment]
@coderabbitai review

CodeRabbit was rate limited on the initial push, so this PR currently has no review coverage. Requesting one now that the branch is current with `main`.

Context for reviewers: this is a follow-up to #898, fixing a cross-node token-boundary case Codex found there after it merged. The guard now asks whether the following sibling *opens* a whitespace-or-end boundary rather than whether it *continues* the command name — the two differ exactly on punctuation, which is what let `["$review", "."]` chip into a structural `review` invocation while the prompt serialized as `/review.`.

File: clients/gui-app/src/lib/composer/tiptap-json-content.ts [hunk 1]
```diff
@@ -273,20 +276,38 @@ function nodesWithLeadingSlashCommandNode(
...
+ * closes no token either.
+ */
+function closesLeadingToken(node: JsonContent | undefined): boolean {
+  // Nothing follows: the token really did end.
+  if (node === undefined) return true;
+  // A hard break serializes to a newline, which is whitespace.
+  if (node.type === "hardBreak") return true;
+  // Anything else that does not open with whitespace - punctuation, more name
+  // characters, a mention chip - leaves the token unterminated.
+  return node.type === "text" && /^\s/.test(node.text ?? "");
```
[the remark this answers is not in the document]
Comment (hdkshingala) [reply clients/gui-app/src/lib/composer/tiptap-json-content.ts]: Confirmed and fixed in `95fa286f`.

Verified the mechanism before changing anything — `plainTextFromNode` returns `""` for `imageAttachment` and `attachmentGroup`, and `plainTextFromNodes` filters empties *before* the join, so `["$frontend-design", imageAttachment, " polish this"]` really does serialize to `/frontend-design polish this`. Refusing there cost the chip its `kind`/`path` for no reason but an atom sitting next to the token.

**I did not add attachments as a fourth case.** That predicate has now been wrong three times — name characters (#898), punctuation (this PR's first round), and now a textless atom — and every time because it re-derived, per node type, something the serializer already knows. So it now asks the serializer:

```ts
function closesLeadingToken(rest: ReadonlyArray<JsonContent>): boolean {
  const text = plainTextFromNodes(rest);
  return text.length === 0 || /^\s/.test(text);
}
```

The token is closed when the remaining siblings render to nothing or to leading whitespace. Attachments drop out for free, a hard break contributes its `\n`, and a mention or another chip contributes text that refuses the boundary — all by construction, and by definition it cannot disagree with what the host's parser receives, which is the property the per-type version could never guarantee.

Call site is guarded on `!state.complete` so the tail is not serialized once per node after the scan settles.

Tests — two new, and the fix keeps all four boundary cases green together:

| case | result |
|---|---|
| `["$fd", img, " polish this"]` | chips |
| `["$fd", img, "."]` | prose |
| `["$fd", "."]` | prose |
| `["$fd", "er the header"]` | prose |

Mutation-probed both directions: restoring the per-node-type predicate reds only the attachment test, and stubbing `closesLeadingToken` to `true` reds all three refusal tests — so the new check is necessary and not merely permissive. 548 composer tests pass.
